### ISSUE TITLE
VirtualBox

### ISSUE THREAD

**Issue by darylheadley** [Installomator/Installomator#902]
virtualbox label downloads Intel version, which fails to install on Apple Silicon

**Comment by darylheadley** [comment]
Changed the script to have two labels for VirtualBox
virtualboxintel and virtualboxsilicon

`virtualboxintel)
    # credit: AP Orlebeke (@apizz)
    name="VirtualBox"
    type="pkgInDmg"
    pkgName="VirtualBox.pkg"
    downloadURL=$(curl -fs "https://www.virtualbox.org/wiki/Downloads" \
        | awk -F '"' "/OSX.dmg/ { print \$4 }")
    appNewVersion=$(curl -fs "https://www.virtualbox.org/wiki/Downloads" | awk -F '"' "/OSX.dmg/ { print \$4 }" | sed -E 's/.*virtualbox\/([0-9.]*)\/.*/\1/')
    expectedTeamID="VB5E2TV963"
    ;;
virtualboxsilicon)
    # credit: Daryl Headley (@darylheadley)
    name="VirtualBox"
    type="pkgInDmg"
    pkgName="VirtualBox.pkg"
    downloadURL=$(curl -fs "https://www.virtualbox.org/wiki/Downloads" \
        | awk -F '"' "/macOSArm64.dmg/ { print \$4 }")
    appNewVersion=$(curl -fs "https://www.virtualbox.org/wiki/Downloads" | awk -F '"' "/macOSArm64.dmg/ { print \$4 }" | sed -E 's/.*virtualbox\/([0-9.]*)\/.*/\1/')
    expectedTeamID="VB5E2TV963"
    ;;`

**Comment by darylheadley** [comment]
I already assume you've got a way of differentiating between architectures and using the same label. Can't see it though

**Comment by darylheadley** [comment]
Found it. Done it. Only tested on Apple Silicon - working.

virtualbox)
    name="VirtualBox"
    type="pkgInDmg"
    pkgName="VirtualBox.pkg"
    if [[ $(arch) == i386 ]]; then
        platform="osx"
    elif [[ $(arch) == arm64 ]]; then
        platform="macOSArm64"
    fi
    downloadURL=$(curl -fs "https://www.virtualbox.org/wiki/Downloads" \
        | awk -F '"' "/"$platform".dmg/ { print \$4 }")
    appNewVersion=$(curl -fs "https://www.virtualbox.org/wiki/Downloads" | awk -F '"' "/"$platform".dmg/ { print \$4 }" | sed -E 's/.*virtualbox\/([0-9.]*)\/.*/\1/')
    expectedTeamID="VB5E2TV963"
    ;;

**Comment by asri-tm** [comment]
curl didn't quite get the correct downloadURL for me

```
% curl -fs "https://www.virtualbox.org/wiki/Downloads" | awk -F '"' "/"OSX".dmg/ { print $4 }"
</li><li><a class="ext-link" href="https://download.virtualbox.org/virtualbox/7.0.6/VirtualBox-7.0.6-155176-OSX.dmg"><span class="icon"> </span>macOS / Intel hosts</a>

% curl -fs "https://www.virtualbox.org/wiki/Downloads" | awk -F '"' "/"macOSArm64".dmg/ { print $4 }"
</li><li><a class="ext-link" href="https://download.virtualbox.org/virtualbox/7.0.6/VirtualBox-7.0.6_BETA4-155176-macOSArm64.dmg"><span class="icon"> </span>Developer preview for macOS / Arm64 (M1/M2) hosts</a>
```

**Comment by scriptingosx** [comment]
merged #904, closing
